src/openMVG/sfm/sfm_data_triangulation.cpp

```diff
@@ -40,12 +40,53 @@ SfM_Data_Structure_Computation_Blind::SfM_Data_Structure_Computation_Blind
 {
 }
 
+/// Triangulate a given track
+bool track_full_triangulation
+(
+  const SfM_Data & sfm_data,
+  const Observations & obs,
+  Vec3 & X
+)
+{
+  if (obs.size() >= 2)
+  {
+    std::vector<Vec3> bearing;
+    std::vector<Mat34> poses;
+    bearing.reserve(obs.size());
+    poses.reserve(obs.size());
+    for (const auto& curr_obs : obs)
+    {
+      const View * view = sfm_data.views.at(curr_obs.first).get();
+      if (!sfm_data.IsPoseAndIntrinsicDefined(view))
+        continue;
+      const IntrinsicBase * cam = sfm_data.GetIntrinsics().at(view->id_intrinsic).get();
+      const Pose3 pose = sfm_data.GetPoseOrDie(view);
+      bearing.emplace_back((*cam)(cam->get_ud_pixel(curr_obs.second.x)));
+      poses.emplace_back(pose.asMatrix());
+    }
+    if (bearing.size() >= 2)
+    {
+      const Eigen::Map<const Mat3X> bearing_matrix(bearing[0].data(), 3, bearing.size());
+      Vec4 Xhomogeneous;
+      TriangulateNViewAlgebraic
+      (
+        bearing_matrix,
+        poses, // Ps are projective cameras.
+        &Xhomogeneous);
+      X = Xhomogeneous.hnormalized();
+      return true;
+    }
+  }
+  return false;
+}
+
+
 /// Triangulate a given track from a selection of observations
 bool track_sample_triangulation
 (
   const SfM_Data & sfm_data,
   const Observations & obs,
-  const std::set<IndexT> & samples,
+  const std::vector<IndexT> & samples,
```

Comment:
We should document that we expect to do not duplicate in the samples array.
`const std::vector<IndexT> & samples, // Unique ids`